File: server/models/events.go
package models

import (
	"time"
)

type Events struct {
	Event_Id    uint64    `json:"event_id" gorm:"primary_key"`
	CreatedAt   time.Time `json:"created_at" gorm:"default:CURRENT_TIMESTAMP"`
	Title       string    `json:"title" gorm:"not null"`
	Description string    `json:"description" gorm:"not null"`
	Event_Date  string    `json:"event_date" gorm:"not null"`
	Location    string    `json:"location" gorm:"not null"`
	Category    string    `json:"category" gorm:"not null"`
}

File: client/src/services/events.ts
import { apiClient } from '../config'

interface CreateEventInput {
  title: string
  description: string
  event_date: string
  location: string
  category: string
}

export const createEvent = (createEventInput: CreateEventInput) => {
  return apiClient.post(`/events/create`, createEventInput)
}

export const getAllEvents = (
  page: number,
  category: string,
  dateRange: string,
) => {
  return apiClient.get(
    `/events?page=${page}&category=${category}&dateRange=${dateRange}`,
  )
}

export const getEventsCount = (category: string, dateRange: string) => {
  return apiClient.get(
    `/events/count?category=${category}&dateRange=${dateRange}`,
  )
}

export const getEventCategories = () => {
  return apiClient.get(`/events/categories`)
}

export const getSingleEvent = (eventId: string) => {
  return apiClient.get(`/events/${eventId}`)
}
